src/p_inter.c

```diff
@@ -216,7 +216,7 @@ boolean P_GiveWeapon(player_t *player, weapontype_t weapon, boolean dropped)
     P_GiveAmmo(player, weaponinfo[weapon].ammo, dropped ? 1 : 2);
 
   return !player->weaponowned[weapon] ?
-    player->weaponowned[player->pendingweapon = weapon] = true : gaveammo;
+    player->weaponowned[player->nextweapon = player->pendingweapon = weapon] = true : gaveammo;
```

Comment:
There are a lot of `player->pendingweapon = weapon` in the codebase. Is there a better way?

Reply:
Nothing immediately comes to mind.